client/src/components/ChangeModal.js

```diff
@@ -43,49 +38,162 @@ const Modal = styled.div`
+  client,
+  clientId,
+  articleId,
+  article,
+  bbd,
+  pzn,
+  ean,
+}) {
   const [{ loading, errormsg }, doPatchClient] = usePatchClient();
-  const [clientname, setClientName] = React.useState(client);
+  const [clientName, setClientName] = React.useState(client);
+  const [articlenumber, setArticleNumber] = React.useState(article);
+  const [bbdValue, setBbdValue] = React.useState(bbd);
+  const [pznValue, setPznValue] = React.useState(pzn);
+  const [eanValue, setEanValue] = React.useState(ean);
+  const [modalTypeClient, setModalTypeClient] = React.useState(false);
+  const [modalTypeArticle, setModalTypeArticle] = React.useState(false);
+  const [{ loadingArticle, errorArticle }, doPatchArticle] = usePatchArticle(
+    articleId
+  );
+  const location = useLocation();
+  function DeclareModalType() {
+    React.useEffect(() => {
+      if (location.pathname === '/articlemaster') {
+        setModalTypeArticle(!modalTypeArticle);
+      }
+
+      if (location.pathname === '/clientmaster') {
+        setModalTypeClient(!modalTypeClient);
+      }
+    }, []);
+  }
+  DeclareModalType();
 
-  async function handleClick() {
-    await doPatchClient(clientId, clientname);
+  async function handleOnClickClient() {
+    await doPatchClient(clientId, clientName);
     close();
   }
 
+  async function handleOnClickArticle() {
+    await doPatchArticle(articleId, articlenumber, client, bbd, pzn, ean);
+    close();
+  }
+
+  function handleClientChange(clientName) {
+    setClientName(clientName);
+  }
   return (
     <>
+      {loadingArticle}
+      {errorArticle && 'Error'}
+      {modalTypeArticle ? (
+        <ModalContainer>
+          <Modal>
+            <CloseImage src={Close} onClick={close}></CloseImage>
+            <Title>Change Article</Title>
+            <ClientDropdown
+              value={clientName}
+              optionTitle="Please Select Client"
+              onContentChange={handleClientChange}
+            ></ClientDropdown>
+            <InputContainer
+              value={articlenumber}
+              maxLength="24"
+              onChange={(event) => {
+                setArticleNumber(event.target.value);
+              }}
+            ></InputContainer>
+            <CheckboxContainer>
+              <Checkbox
+                checkStatus={bbd ? 'checked' : null}
+                option="Best-Before-Date"
+                checked={() => {
+                  setBbdValue(!bbdValue);
+                }}
+              ></Checkbox>
+              <Checkbox
+                checkStatus={pzn ? 'checked' : null}
+                option="PZN Number"
+                checked={() => {
+                  setPznValue(!pznValue);
+                }}
+              ></Checkbox>
+              <Checkbox
+                checkStatus={ean ? 'checked' : null}
+                option="EAN Number"
+                checked={() => {
+                  setEanValue(!eanValue);
+                }}
+              ></Checkbox>
```

Comment:
I know checkbox from HTML:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input/checkbox
https://reactjs.org/docs/forms.html#handling-multiple-inputs

I can use it like this:
`<input type="checkbox" checked={checked} onChange={(event) => setChecked(event.target.checked)} >`

Your checkbox is different, which might be confusing for other developers.

1: `checked` is equal to `onChange`
2: `checkedStatus` is equal to `checked`

Better not change the meaning of these properties:

```suggestion
              <Checkbox
                checked={ean}
                option="EAN Number"
                onChange={(event) => {
                  setEanValue(event.target.checked);
                }}
              />
```

You will reduce the complexity of your code.